Prog/New_Project_Geocode.sas

```diff
@@ -0,0 +1,565 @@
+  Proj_lon = sum( Proj_lon, MAR_LONGITUDE );
+  
+  if image_url ~= "" and Proj_image_url = "" then Proj_image_url = image_url;
+
+  if Streetview_url ~= "" and Proj_streetview_url = "" then Proj_streetview_url = Streetview_url;
+
+  if Bldg_count = 1 then Proj_addre = MAR_MATCHADDRESS;
+  else if Bldg_count <= &MAX_PROJ_ADDRE then Proj_addre = trim( Proj_addre ) || "; " || MAR_MATCHADDRESS;
+  else if Bldg_count = %eval( &MAX_PROJ_ADDRE + 1 ) then Proj_addre = trim( Proj_addre ) || "; others";
+    
+  if last.nlihc_id then do;
+  
+    Proj_x = Proj_x / Bldg_count;
+    Proj_y = Proj_y / Bldg_count;
+    Proj_lat = Proj_lat / Bldg_count;
+    Proj_lon = Proj_lon / Bldg_count;
+    
+    output work.Project_geocode_a;
+
+  end;
+  
+  label
+    Ward2012x = "Ward (2012)"
+    Ssl_std = "Property identification number (square/suffix/lot)"
+    Proj_Name = "Project name"
+    NLIHC_ID = "Preservation Catalog project ID"
+    address_id = "MAR address ID"
+    streetview_url = "Google Street View URL"
+    Anc2012 = "Advisory Neighborhood Commission (2012)"
+    Psa2012 = "Police Service Area (2012)"
+    Geo2010 = "Full census tract ID (2010): ssccctttttt"
+    Cluster_tr2000 = "Neighborhood cluster (tract-based, 2000)"
+    Cluster_tr2000_name = "Neighborhood cluster names (tract-based, 2000)"
+    zip = "ZIP code (5 digit)"
+    image_url = "OCTO property image URL"
+    Proj_addre = "Project address"
+    Proj_address_id = "Project MAR address ID"
+    Proj_image_url = "OCTO property image URL"
+    Proj_lat = "Project latitude"
+    Proj_lon = "Project longitude"
+    Proj_streetview_url = "Google Street View URL"
+    Proj_x = "Project longitude (MD State Plane Coord., NAD 1983 meters)"
+    Proj_y = "Project latitude (MD State Plane Coord., NAD 1983 meters)"
+    Proj_zip = "ZIP code (5 digit)"
+    Bldg_addre = "Building address"
+    Bldg_x = "Building longitude (MD State Plane Coord., NAD 1983 meters)"
+    Bldg_y = "Building latitude (MD State Plane Coord., NAD 1983 meters)"
+    Bldg_lon = "Building longitude"
+    Bldg_lat = "Building latitude"
+    Bldg_count = "Number of buildings for project";
+  
+  format Ward2012x $ward12a.;
+  
+  rename Ward2012x = Ward2012;
+  drop Ward2012;
+  
+run;
+
+**remove projects from geocode datasets that are no longer in project dataset**;
+
+proc sql;
+
+   create table project as
+   select *
+   from prescat.project_geocode
+   where nlihc_id in (select distinct nlihc_id from prescat.project)
+   ;
+
+quit;
+
+proc sql;
+
+   create table building as
+   select *
+   from prescat.building_geocode
+   where nlihc_id in (select distinct nlihc_id from prescat.project)
+   ;
+
+quit;
+
+**compare new datasets with old geocode datasets**;
+
+proc compare base=prescat.project_geocode compare=work.project listall maxprint=(40,32000);
+  id nlihc_id;
+run;
+
+proc compare base=prescat.building_geocode compare=work.building listall maxprint=(40,32000);
+  id Bldg_address_id;
+run;
```

Comment:
Good to use Proc Compare to check things. :+1:  But what are we looking for exactly? Will this be part of the permanent process, or just a checking things for now? If permanent, what should the person running the program be looking for?